api/routes/api_key.go

```diff
@@ -0,0 +1,98 @@
+package routes
+
+import (
+	"net/http"
+
+	"github.com/shellhub-io/shellhub/api/pkg/gateway"
+	"github.com/shellhub-io/shellhub/pkg/api/requests"
+)
+
+const (
+	CreateAPIKeyURL = "/namespaces/:tenant/api-key"
+	RenameAPIKeyURL = "/namespaces/:tenant/api-key"
+	GetAPIKeyURL    = "/namespaces/:tenant/api-key"
+	DeleteAPIKeyURL = "/namespaces/:tenant/api-key"
+)
+
+func (h *Handler) CreateAPIKey(c gateway.Context) error {
+	var req requests.CreateAPIKey
+	if err := c.Bind(&req); err != nil {
+		return err
+	}
+
+	if err := c.Validate(&req); err != nil {
+		return err
+	}
+
+	var userID string
+	if v := c.ID(); v != nil {
+		userID = v.ID
+	}
+
+	apiKey := c.Request().Header.Get("X-API-KEY")
+
+	token, err := h.service.CreateAPIKey(c.Ctx(), userID, req.Name, apiKey, req.ExpiresAt)
+	if err != nil {
+		return err
+	}
+
+	return c.JSON(http.StatusOK, map[string]interface{}{
+		"token": token,
+	})
+}
+
+func (h *Handler) GetAPIKey(c gateway.Context) error {
+	var req requests.UserParam
```

Comment:
I believe there's a fatal issue here. You need to create a new `requests` struct, which should include the appropriate bind tags. Specifically, you'll need to bind the tenant ID, page, and sorter fields.

Reply:
changed requests to receive the appropriate parameters